org.opentosca.bus.management.deployment.plugin.remote/src/org/opentosca/bus/management/deployment/plugin/remote/ManagementBusDeploymentPluginRemote.java

```diff
@@ -0,0 +1,181 @@
+ *
+ */
+public class ManagementBusDeploymentPluginRemote implements IManagementBusDeploymentPluginService {
+
+    static final private Logger LOG = LoggerFactory.getLogger(ManagementBusDeploymentPluginRemote.class);
+
+    @Override
+    public Exchange invokeImplementationArtifactDeployment(final Exchange exchange) {
+
+        ManagementBusDeploymentPluginRemote.LOG.debug("Trying to deploy IA on remote OpenTOSCA Container.");
+        final Message message = exchange.getIn();
+
+        // perform remote deployment
+        final Exchange response = sendRequestToRemoteContainer(message, RemoteOperations.invokeIADeployment);
+
+        // extract the endpoint URI from the response
+        final URI endpointURI = response.getIn().getHeader(MBHeader.ENDPOINT_URI.toString(), URI.class);
+        ManagementBusDeploymentPluginRemote.LOG.debug("Result of remote deployment: Endpoint URI: {}", endpointURI);
+
+        // add the header to the incoming exchange and return result
+        message.setHeader(MBHeader.ENDPOINT_URI.toString(), endpointURI);
+        return exchange;
+    }
+
+    @Override
+    public Exchange invokeImplementationArtifactUndeployment(final Exchange exchange) {
+
+        ManagementBusDeploymentPluginRemote.LOG.debug("Trying to undeploy IA on remote OpenTOSCA Container.");
+        final Message message = exchange.getIn();
+
+        // perform remote undeployment
+        final Exchange response = sendRequestToRemoteContainer(message, RemoteOperations.invokeIAUndeployment);
+
+        // extract the undeployment state from the response
+        final boolean state = response.getIn().getHeader(MBHeader.OPERATIONSTATE_BOOLEAN.toString(), boolean.class);
+        ManagementBusDeploymentPluginRemote.LOG.debug("Result of remote undeployment: Success: {}", state);
+
+        // add the header to the incoming exchange and return result
+        message.setHeader(MBHeader.OPERATIONSTATE_BOOLEAN.toString(), state);
+        return exchange;
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public List<String> getSupportedTypes() {
+
+        // This plug-in supports only the special type 'remote' which is used to forward deployment
+        // requests to other OpenTOSCA Containers.
+        final List<String> types = new ArrayList<>();
+        types.add(Constants.REMOTE_TYPE);
+
+        return types;
+    }
+
+    @Override
+    /**
+     * {@inheritDoc}
+     */
+    public List<String> getCapabilties() {
+
+        // This plug-in is intended to move deployment requests from one OpenTOSCA Container to
+        // another one. At the destination OpenTOSCA Container the deployment is done by one of the
+        // other available deployment plug-ins. Therefore, it has to be checked if this other
+        // deployment plug-in provides all needed capabilities before moving the request to the
+        // other Container. So, as this plug-in is only a redirection it does not provide any
+        // capabilities.
```

Comment:
This seems like it should be the javadoc instead of a comment inside the implementation